src/rewriter/node.py

```diff
@@ -18,142 +18,181 @@
 
     # Bitwise bit-vector operations
-    BVNOT = auto()
-    BVAND = auto()
-    BVOR = auto()
-    BVXOR = auto()
-    BVNAND = auto()
-    BVNOR = auto()
-    BVXNOR = auto()
-
-    CONCAT = auto()
-
-    BVITE = auto()
-    BVCOMP = auto()
-
-    BVCONST = auto()
-    ZERO_EXTEND = auto()
-    SIGN_EXTEND = auto()
-    EXTRACT = auto()
-    REPEAT = auto()
+    BVNOT = ('bvnot', 'BITVECTOR_NOT')
+    BVAND = ('bvand', 'BITVECTOR_AND')
+    BVOR = ('bvor', 'BITVECTOR_OR')
+    BVXOR = ('bvxor', 'BITVECTOR_XOR')
+    BVNAND = ('bvnand', 'BITVECTOR_NAND')
+    BVNOR = ('bvnor', 'BITVECTOR_NOR')
+    BVXNOR = ('bvxnor', 'BITVECTOR_XNOR')
+    BVUADDO = ('bvuaddo', 'BITVECTOR_UADDO')
+    BVSADDO = ('bvsaddo', 'BITVECTOR_SADDO')
+    BVUMULO = ('bvumulo', 'BITVECTOR_UMULO')
+    BVSMULO = ('bvsmulo', 'BITVECTOR_SMULO')
+    BVUSUBO = ('bvusubo', 'BITVECTOR_USUBO')
+    BVSSUBO = ('bvssubo', 'BITVECTOR_SSUBO')
+    BVSDIVO = ('bvsdivo', 'BITVECTOR_SDIVO')
+
+    BVITE = ('bvite', 'BITVECTOR_ITE')
+    BVCOMP = ('bvcomp', 'BITVECTOR_COMP')
+
+    BVCONST = ('bv', 'CONST_BITVECTOR_SYMBOLIC')
+    ZERO_EXTEND = ('zero_extend', 'BITVECTOR_ZERO_EXTEND')
+    SIGN_EXTEND = ('sign_extend', 'BITVECTOR_SIGN_EXTEND')
+    CONCAT = ('concat', 'BITVECTOR_CONCAT')
+    EXTRACT = ('extract', 'BITVECTOR_EXTRACT')
+    REPEAT = ('repeat', 'BITVECTOR_REPEAT')
 
     ###########################################################################
     # Boolean
     ###########################################################################
 
-    NOT = auto()
-    AND = auto()
-    OR = auto()
-    IMPLIES = auto()
-    XOR = auto()
+    NOT = ('not', 'NOT')
+    AND = ('and', 'AND')
+    OR = ('or', 'OR')
+    IMPLIES = ('=>', 'IMPLIES')
+    XOR = ('xor', 'XOR')
 
     ###########################################################################
     # Arithmetic
     ###########################################################################
 
-    NEG = auto()
-    ADD = auto()
-    SUB = auto()
-    MULT = auto()
-    INT_DIV = auto()
-    DIV = auto()
-    MOD = auto()
-    ABS = auto()
-    LT = auto()
-    GT = auto()
-    LEQ = auto()
-    GEQ = auto()
+    NEG = (None, 'NEG')  # This is parsed with SUB, so the key is None
+    ADD = ('+', 'ADD')
+    SUB = ('-', 'SUB')
+    MULT = ('*', 'MULT')
+    INT_DIV = ('div', 'INTS_DIVISION')
+    DIV = ('/', 'DIVISION')
+    MOD = ('mod', 'INTS_MODULUS')
+    ABS = ('abs', 'ABS')
+    LT = ('<', 'LT')
+    GT = ('>', 'GT')
+    LEQ = ('<=', 'LEQ')
+    GEQ = ('>=', 'GEQ')
+    POW2 = ('int.pow2', 'POW2')
 
     ###########################################################################
     # Theory-independent
     ###########################################################################
 
-    EQ = auto()
-    ITE = auto()
-    LAMBDA = auto()
-    BOUND_VARS = auto()
+    EQ = ('=', 'EQUAL')
+    ITE = ('ite', 'ITE')
+    LAMBDA = (None, 'LAMBDA')  # Special parser backdoor
```

Comment:
What does this comment mean?

Reply:
Lambda is not actually an operator and has no name

Reply:
Ok, then clarify the comment since "Special parser backdoor" is not really informative.

Reply:
fixed